### PR TITLE
Fix crash in Style/ExplicitBlockArgument

### PR DESCRIPTION
**Pull request by ghiculescu**
Fixes https://github.com/rubocop-hq/rubocop/issues/8825

### CODE REVIEW

**Comment by koic** [comment]
Nice! Can you add the changelog entry to the bug fixes section of CHANGELOG.md and add your account to the bottom of the changelog as `[@ghiculescu]: https://github.com/ghiculescu`.?
https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 * [#8809](https://github.com/rubocop-hq/rubocop/pull/8809): Fix multiple offense detection for `Style/For`. ([@pbernays][])
 * [#8801](https://github.com/rubocop-hq/rubocop/issues/8801): Fix `Layout/SpaceAroundEqualsInParameterDefault` only registered once in a line. ([@rdunlop][])
 * [#8514](https://github.com/rubocop-hq/rubocop/issues/8514): Correct multiple `Style/MethodDefParentheses` per file. ([@rdunlop][])
+* [#8826](https://github.com/rubocop-hq/rubocop/issues/8826): Fix crash in `Style/ExplicitBlockArgument`. ([@ghiculescu][])
```
Comment (koic) [inline CHANGELOG.md:15]: Can you update the changelog entry and squash your commits into one?

```suggestion
* [#8825](https://github.com/rubocop-hq/rubocop/issues/8825): Fix crash in `Style/ExplicitBlockArgument` when code is called outside of a method. ([@ghiculescu][])
```

**Comment by ghiculescu** [comment]
I can squash, but out of curiosity why not use the github merge option that squashes for you?

**Comment by koic** [comment]
Great thanks!

> I can squash, but out of curiosity why not use the github merge option that squashes for you?

Of course I can use squash merge, but personally I'd like to create a merge commit if possible. Because changing the commit hash can increase the cost of tracing for maintenance. There are several different options of thinking about this, and I don't want to spend as time on this subject as possible. Anyway, I'm glad that it will be improved. Thank you for taking your time 😅

**Pull request merged**